apps/qubit/modules/settings/actions/diacriticsAction.class.php

```diff
@@ -0,0 +1,88 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+class SettingsDiacriticsAction extends SettingsEditAction
+{
+    // Arrays not allowed in class constants
+    public static $NAMES = [
+        'diacritics',
+    ];
+
+    public function earlyExecute()
+    {
+        parent::earlyExecute();
+
+        $this->updateMessage = $this->i18n->__('Diacritics settings saved.');
+
+        $this->settingDefaults = [
+            'diacritics' => 0,
+        ];
+    }
+
+    public function execute($request)
+    {
+        parent::execute($request);
+    }
+
+    public function uploadDiacritics($request)
+    {
+        $file = $request->getFiles('mappings');
+        $diacriticsMappingPath =
+            sfConfig::get('sf_plugins_dir') . DIRECTORY_SEPARATOR .
+            'arElasticSearchPlugin' . DIRECTORY_SEPARATOR .
+            'config' . DIRECTORY_SEPARATOR .
+            'diacritics_mapping.yml';
+
+        if (1 === (int) $request->getPostParameters()['diacritics']) {
+            if ('application/x-yaml' === $request->getFiles()['mappings']['type']) {
+                // Move uploaded file to replace diacritics_mapping.yml.
+                try {
+                    $file = Qubit::moveUploadFile($file);
+                } catch (sfException $e) {
+                    $this->getUser()->setFlash('error', $e->getMessage());
+                }
+
+                // Replace /config/diacritics_mapping.yml with uploaded file
+                $uploadMappingPath = $file['tmp_name'];
+                $diacriticsMappings = file_get_contents($uploadMappingPath);
+                file_put_contents($diacriticsMappingPath, $diacriticsMappings);
+                $this->getUser()->setFlash('notice', $this->context->i18n->__('Diacritics setting updated. Rebuild the search index for the changes to be applied.'));
+            } else {
+                QubitSetting::findAndSave('diacritics', 0, ['sourceCulture' => true]);
+                $this->getUser()->setFlash('error', $this->context->i18n->__('Error: Unable to upload diacritics mapping.'));
+            }
+        } else {
+            // Reset diacritics yaml when disabling the setting
+            unlink($diacriticsMappingPath);
+        }
+    }
+
+    protected function addField($name)
+    {
+        switch ($name) {
+            case 'diacritics':
+                $this->form->setDefault($name, $this->settingDefaults[$name]);
+                $this->form->setWidget($name, new sfWidgetFormSelectRadio(['choices' => [0 => $this->i18n->__('Disabled'), 1 => $this->i18n->__('Enabled')]], ['class' => 'radio']));
+                $this->form->setValidator($name, new sfValidatorChoice(['choices' => [1, 0]]));
+                $this->form->setWidget('mappings', new sfWidgetFormInputFile());
```

Comment:
It would be nice to set the accept value: https://github.com/artefactual/atom/blob/qa/2.x/apps/qubit/modules/informationobject/actions/uploadFindingAidAction.class.php#L57